W-18518517 A2A Connector (#2906)
mulesoft/docs-connectors: a2a/1.0/modules/ROOT/pages/index.adoc

= A2A Connector 0.1.0-BETA 
:lc-connector-name: a2a

Anypoint Connector for Agent2Agent (A2A) (A2A Connector) facilitates inter-agent communication for Large Language Model (LLM) agents and other applications by implementing the A2A protocol. This enables agents to interact and delegate specialized tasks to other agents.

The A2A protocol is an open standard that supports:

* Capability discovery
+

> Identifies the functions and services offered by agents

Reviewer: ```suggestion
Identifies the functions and services offered by agents.
```

* Interaction negotiation
+
Establishes communication methods between agents.
* Secure information exchange
+ 
Transmits data between agent systems without requiring direct internal access.

A2A Connector provides Mule app integration that functions as both A2A servers and A2A clients.

[NOTE]
====

> Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 

Reviewer: ```suggestion
Anypoint Connector for Agent2Agent (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 
```

> Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes https://docs.mulesoft.com/general/usage-metrics[Mule messages] and is at the Customer's sole discretion. 

Reviewer: ```suggestion
Anypoint Connector for Agentforce (A2A) is a beta connector that is subject to the Beta Services Terms at https://www.salesforce.com/company/legal/agreements/[Agreements - Salesforce.com] or a written Unified Pilot Agreement if executed by Customer, and the Non-GA Gen AI and Non-GA Credit Consumption terms in the https://ptd.salesforce.com/?_ga=2.247987783.1372150065.1709219475-629000709.1639001992[Product Terms Directory]. Use of this beta connector with external systems or services consumes xref:general::usage-metrics.adoc[Mule messages] and is at the Customer's sole discretion. 
```

Reviewer: Not sure if this should be an xref

Author: Yeah. The PM wanted me to add it.

====

For information about compatibility and fixed issues, see the A2A Connector Release Notes.

> == Before You Begin

Reviewer 2: Can we add somewhere that the following A2A Capabilities are currently not supported in this BETA release?

```
Both Client And Server currently do not support
1. Streaming
2. Notifications
3. Task History
```

To use this connector, you must be familiar with:

* Anypoint Connectors
* Mule runtime engine (Mule)
* Elements and global elements in a Mule flow
* How to create a Mule app using Anypoint Code Builder or Anypoint Studio

Before creating an app, you must have:

* Java 17
* Apache Maven
* Anypoint Platform
* Credentials to connect with the target resource
* The latest versions of Anypoint Code Builder or Anypoint Studio

== Common Use Cases for the Connector

Build specialized domain specific agents each for Customer Service, Sales, or Marketing and have other agents interact with these agents using the A2A protocol.

== Connection Types

A2A Connector supports these connection types:

* A2A server

> ** Introduces a new A2A Listener with a *Task Listener* operation.

Reviewer: ```suggestion
** Introduces a new A2A Listener with a *Task Listener* source.
```
